wallets/provider-walletconnect-2/src/index.ts

```diff
@@ -149,17 +161,27 @@ export const switchNetwork: SwitchNetwork = async ({
   network,
   instance,
   meta,
+  getState,
 }) => {
-  /**
-   * Wallet connect is a multichain protocol and we can not determine the connected wallet
-   * supports which wallet, `extend`ing session doesn't work during a bug in their utils packages.
-   * So we will try to make a new session with `network` that user needs to switch.
-   */
-  const session = await trySwitchByCreatingNewSession(instance, {
-    network,
-    meta,
-  });
-  instance.session = session;
+  const isAsyncSwitchNetork = needSessionRecreateOnSwitchNetwork(instance);
```

Comment:
you can directly call your function (as `avoidUpdateOnSwitchNetwork`) here as well.